test-projects/social-login/server.js

```diff
@@ -0,0 +1,105 @@
+const keystone = require('@keystone-alpha/core');
+const { endAuthedSession } = require('@keystone-alpha/session');
+const {
+  facebookAuthEnabled,
+  githubAuthEnabled,
+  twitterAuthEnabled,
+  googleAuthEnabled,
+  wpAuthEnabled,
+  port,
+  staticRoute,
+  staticPath,
+} = require('./config');
+const {
+  configureFacebookAuth,
+  configureGitHubAuth,
+  configureGoogleAuth,
+  configureTwitterAuth,
+  configureWPAuth,
+  setupAuthRoutes,
+  InitializePassportAuthStrategies,
+} = require('./auth');
+
+const initialData = require('./data');
+keystone
+  .prepare({ port })
+  .then(async ({ server, keystone: keystoneApp }) => {
+    const socialLogins = [];
+    if (facebookAuthEnabled) {
+      socialLogins.push(configureFacebookAuth(keystoneApp));
+    }
+
+    if (githubAuthEnabled) {
+      socialLogins.push(configureGitHubAuth(keystoneApp));
+    }
+
+    if (twitterAuthEnabled) {
+      socialLogins.push(configureTwitterAuth(keystoneApp));
+    }
+
+    if (googleAuthEnabled) {
+      socialLogins.push(configureGoogleAuth(keystoneApp));
+    }
+
+    if (wpAuthEnabled) {
+      socialLogins.push(configureWPAuth(keystoneApp));
+    }
+
+    await keystoneApp.connect();
+
+    socialLogins.forEach(strategy => setupAuthRoutes({ strategy, server }));
+
+    if (
+      facebookAuthEnabled ||
+      githubAuthEnabled ||
+      googleAuthEnabled ||
+      twitterAuthEnabled ||
+      wpAuthEnabled
+    ) {
+      InitializePassportAuthStrategies(server.app);
```

Comment:
I believe this block should be moved above `setupAuthRoutes`, as `passport.initialize()` should come before other middlewares.